specs/Menu.md

```diff
@@ -1,731 +1,545 @@
+const onHideContextualMenu = () => setShowContextualMenu(false);
+
+<ContextualMenu
+  hidden={!showContextualMenu}
+  onItemClick={onHideContextualMenu}
+  onDismiss={onHideContextualMenu
+/>;
 
-### AntDesign DOM Structure
-
-_Note:_ Some long class names removed
-
-```HTML
-<ul role="menu">
-   <li class="ant-menu-item" role="menuitem" style="padding-left: 24px;">Item one</li>
-   <li class=" ant-menu-item-divider"></li>
-   <li class="ant-menu-item" role="menuitem" style="padding-left: 24px;"> Item two</li>
-   <li class="ant-menu-submenu ant-menu-submenu-inline" role="menuitem">
-      <div class="ant-menu-submenu-title" aria-expanded="false" aria-haspopup="true" style="padding-left: 24px;">
-          <span>Item Three</span>
-          <i class="ant-menu-submenu-arrow"/>
-      </div>
-   </li>
-</ul>
+// v0 uncontrolled Popup
+const [open, setOpen] = React.useState(false);
+
+<Popup
+  onOpenChange={(e, props: PopupProps) => {/*react on changes*/}}
+  trigger={<Button icon={<OpenOutsideIcon />} title="Open popup" />}
+/>;
+
+// v0 controlled Popup - used with trigger disables autocontrol
+const [open, setOpen] = React.useState(false);
+
+<Popup
+  open={open}
+  onOpenChange={(e, props: PopupProps) => setOpen(!props.open)}
+  trigger={<Button icon={<OpenOutsideIcon />} title="Open popup" />}
+/>;
 ```
 
-## Recommendations
-
-### Recommended DOM Structure
-
-```HTML
-<div role="menu">
-    <div role="menuitem">{"Item one"}</div>
-    <div role="presentation" />
-    <div role="menuitem">{"Item two"}</div>
-    <a role="menuitem" href="www.isALink.com"> link1 </a>
-    <div role="menuitem">
-        {"Item Three"}
-    </div>
-        <!-->
-            I'm not sure if this is the right role or if it is actually a menu item. It should
-            probably be related to its parent's item in some way.
-        <-->
-    <div role="presentation">
-        <!-->
-            Note: Does not necessarily need to be part of the same dom
-            tree, could be floating.
-        <-->
-        <div role="menu">
-            <div role="menuitem">{"SubItem one"}</div>
-            <div role="menuitem">{"SubItem two"}</div>
-        </div>
-    </div>
-</ul>
+### Keyboard navigation
+
+Both v7 and v0 support arrow navigation in the menu, and home/end keys to jump to first and last items respectively.
+
+One interesting difference is that the v7 `ContextualMenu` will also tab through items. The v0 `Menu` on the other hand uses tab to focus in/out of the entire component.
+
+`ContextualMenu` will also allow disabled items to be focusable while navigating the list, while the v0 `Menu` does not permit this.
+
+### Selection state
+
+The v0 `Menu` component supports and `active` state and has a number of props to manage this state. However, this state only affects items visually and does not perform the same function as menu item checkboxes or radio items. The `active` state of menu items can both be controlled and autocontrolled
```

Comment:
The active state is confusing and should be removed I think - given that Tabs and Toolbar will be a separate component.

Reply:
however there are edge cases where menu item checkbox and menu item radio might have active state

Reply:
I agree, I was super confused trying to figure out in our own codebase what this `active` state actually was.

Reply:
I make no mention of this in the proposed behaviour/api so far and do not intend to